File: app/src/main/java/com/amarnehsoft/holyquran/db/AyaDao.java
package com.amarnehsoft.holyquran.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.amarnehsoft.holyquran.model.Aya;
import com.amarnehsoft.holyquran.model.Surah;

import java.util.List;

import io.reactivex.Maybe;
import io.reactivex.Single;

@Dao
public interface AyaDao {
    @Query("select * from aya where number =:number")
    Aya getByNumber(int number);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Aya aya);
}

File: app/src/main/java/com/amarnehsoft/holyquran/db/AyaDao.kt
package com.amarnehsoft.holyquran.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface AyaDao {

    @Query("select * from ayahentity")
    suspend fun getAll(): List<AyahEntity>

    @Query("select * from ayahentity where number =:number")
    suspend fun getByNumber(number: Int): AyahEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(aya: AyahEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(list: List<AyahEntity>)
}
